nerfstudio/data/utils/data_utils.py

```diff
@@ -31,7 +63,7 @@ def get_image_mask_tensor_from_path(filepath: Path, scale_factor: float = 1.0) -
         width, height = pil_mask.size
         newsize = (int(width * scale_factor), int(height * scale_factor))
         pil_mask = pil_mask.resize(newsize, resample=Image.Resampling.NEAREST)
-    mask_tensor = torch.from_numpy(np.array(pil_mask)).unsqueeze(-1).bool()
+    mask_tensor = torch.from_numpy(pil_to_numpy(pil_mask)).unsqueeze(-1).bool()
```

Comment:
what was wrong with like `numpy.asarray(pil_image)` or can we maybe use https://pytorch.org/vision/main/generated/torchvision.transforms.functional.pil_to_tensor.html  ?   `pil_to_numpy()` looks odd.. tho i understand there's PIL resize() for some reason, i wonder if the resize could actually not use PIL (nerfstudio uses cv2.resize() a lot ) ...  not sure why the orig authors chose PIL resize

Reply:
After some digging + discussion, doing `asarray` leads an unwriteable numpy array, which could have conflicts when converting to Pytorch tensors described [here](https://github.com/pytorch/pytorch/blob/07adae3dac85d51303e52bbaaa3d45df6da54ae5/torch/csrc/utils/tensor_numpy.cpp#L205)